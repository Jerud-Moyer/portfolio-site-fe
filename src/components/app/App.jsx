import React from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch
} from 'react-router-dom';
import FactDetail from '../facts/FactDetail';
import AdminPage from '../admin/AdminPage';
import styles from './app.css';
import UpdateFact from '../admin/UpdateFact';
import CreateFact from '../admin/CreateFact';
import Portfolio from '../portfolio/Portfolio';
import FactListPage from '../pages/FactListPage';


export default function App() {
  return (
    <div className={styles.view}>
      <Router>
        <Switch>
          <Route exact path="/" component={Portfolio} />
          <Route exact path="/list" component={FactListPage} />
          <Route exact path="/detail/:id" component={FactDetail} />
          <Route exact path="/admin" component={AdminPage} />
          <Route exact path="/UpdateFact/:id" component={UpdateFact} />
          <Route exact path="/Create" component={CreateFact} />
        </Switch>
      </Router>
    </div>
  );
}
