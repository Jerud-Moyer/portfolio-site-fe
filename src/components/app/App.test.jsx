
import React from 'react';
import { Provider } from 'react-redux';
import { render, cleanup } from '@testing-library/react';
import App from './App';
import store from '../../store';
import { MemoryRouter } from 'react-router-dom';
import { mockAllIsIntersecting } from 'react-intersection-observer/test-utils';

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(
      <Provider store={store} >
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </Provider>
    );
    mockAllIsIntersecting(true);
    expect(asFragment()).toMatchSnapshot();
  });
}
);
