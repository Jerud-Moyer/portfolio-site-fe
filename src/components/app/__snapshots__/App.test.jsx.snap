// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <div
    class="view"
  >
    <div
      class="portfolio"
    >
      <nav
        data-testid="nav-bar"
      >
        <div
          class="navBarOuter"
        >
          <nav
            class="navBar"
          >
            <a
              href="#home"
            >
              home
            </a>
            <a
              href="#projects"
            >
              projects
            </a>
            <a
              href="#freelance"
            >
              freelance
            </a>
            <a
              href="#tech"
            >
              tech stack
            </a>
            <a
              href="#about"
            >
              about me
            </a>
            <a
              href="/moreProjects"
            >
              more projects
            </a>
          </nav>
          <div
            class="mobileNav"
          >
            <nav
              class="hamburger"
            >
              =
            </nav>
          </div>
        </div>
      </nav>
      <div
        class="sectionZero"
      >
        <div
          class="homePage"
          id="home"
        >
          <div
            class="topBox"
          >
            <div
              class="infoBox"
            >
              <h1
                class="headline"
              >
                Jerud Moyer
              </h1>
              <div
                class="bulletBox"
              >
                <ul>
                  <li
                    class="point0"
                  >
                    Full-Stack Software Developer
                  </li>
                  <li
                    class="point1"
                  >
                    Problem Solver
                  </li>
                  <li
                    class="point2"
                  >
                    Creator
                  </li>
                  <li
                    class="point3"
                  >
                    Builder
                  </li>
                </ul>
              </div>
            </div>
            <div
              class="contactBox"
            >
              <div
                class="elementBox"
              >
                <p
                  class="invisible"
                >
                   contact me =&gt;
                </p>
                <div
                  class="popUpInner"
                >
                  <div>
                    <div
                      class="popUp"
                    >
                      <label
                        class="label"
                        for="form-checkboxhome-page"
                      >
                        <img
                          alt="email"
                          class="pic"
                          src="/path/image"
                        />
                      </label>
                      <input
                        class="toggle"
                        id="form-checkboxhome-page"
                        type="checkbox"
                      />
                      <div
                        class="contain"
                      >
                        <div />
                      </div>
                      <div
                        class="form"
                      >
                        <div>
                          <div>
                            <div
                              class="contact"
                            >
                              <label
                                class="Xbox"
                                for="form-checkboxhome-page"
                              >
                                X
                              </label>
                              <input
                                class="checkbox"
                                id="form-checkboxhome-page"
                                type="checkbox"
                              />
                              <form
                                id="contact-formhome-page"
                                method="POST"
                              >
                                <input
                                  id="namehome-page"
                                  name="name"
                                  placeholder="name required"
                                  type="text"
                                  value=""
                                />
                                <input
                                  id="emailhome-page"
                                  name="email"
                                  placeholder="email required"
                                  type="text"
                                  value=""
                                />
                                <textarea
                                  id="messagehome-page"
                                  name="message"
                                  placeholder="message required"
                                />
                                <button>
                                  submit
                                </button>
                              </form>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="elementBox"
              >
                <p
                  class="invisible"
                >
                  my resume =&gt;
                </p>
                <a
                  href="/assets/jerudresume.pdf"
                >
                  <img
                    alt="PDF"
                    class="mediaPic"
                    src="/path/image"
                  />
                </a>
              </div>
              <div
                class="elementBox"
              >
                <p
                  class="invisible"
                >
                  Twitter =&gt;
                </p>
                <a
                  href="https://twitter.com/Jerud7"
                >
                  <img
                    alt="Twitter"
                    class="mediaPic"
                    src="/path/image"
                  />
                </a>
              </div>
              <div
                class="elementBox"
              >
                <p
                  class="invisible"
                >
                  LinkedIn =&gt;
                </p>
                <a
                  href="https://www.linkedin.com/in/jerud-moyer/"
                >
                  <img
                    alt="LinkedIn"
                    class="mediaPic"
                    src="/path/image"
                  />
                </a>
              </div>
              <div
                class="elementBox"
              >
                <p
                  class="invisible"
                >
                  Github =&gt;
                </p>
                <a
                  href="https://github.com/Jerud-Moyer"
                >
                  <img
                    alt="GitHub"
                    class="mediaPic"
                    src="/path/image"
                  />
                </a>
              </div>
            </div>
          </div>
          <div
            class="factBox"
          >
            <div
              class="fact"
            >
              <a
                class="link"
                href="/detail/undefined"
              >
                <p
                  class="undefined"
                />
              </a>
            </div>
          </div>
          <div
            class="arrowBox"
          >
            <div
              class="arrowBox"
            >
              <div
                class="arrow"
              >
                ï¸¾
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sectionOne"
      >
        <div
          class="projectsPage"
          id="projects"
        >
          <div
            class="listBox1"
          >
            <div
              class="projectBox"
              style="transition: ease-in; visibility: visible;"
            >
              <div
                class="listBox"
              >
                <h1
                  class="headline"
                >
                  Projects
                </h1>
                <ul
                  class="projectList"
                >
                  <li>
                    <div
                      class="projectItem"
                    >
                      <h2
                        class="name"
                      >
                        Mood-Music
                      </h2>
                      <img
                        alt="Mood-Music"
                        class="projectPicInView"
                        src="/assets/moodmusic.png"
                      />
                      <p
                        class="description"
                      >
                        Take a selfie, recieve an emotional analysis, then submit for a spotify widget complete with a playlist to match your mood!
                      </p>
                      <a
                        class="siteLink"
                        href="https://mood-music.netlify.app/"
                      >
                        Visit Mood-Music
                      </a>
                      <a
                        class="gitLink"
                        href="https://github.com/Silver-Surfers"
                      >
                        Git Repo
                      </a>
                    </div>
                  </li>
                  <li>
                    <div
                      class="projectItem"
                    >
                      <h2
                        class="name"
                      >
                        Project Vobot
                      </h2>
                      <img
                        alt="Project Vobot"
                        class="projectPicInView"
                        src="/assets/vobot.png"
                      />
                      <p
                        class="description"
                      >
                        Tweet or text the name of your state and recieve voter information in return.
                      </p>
                      <a
                        class="siteLink"
                        href="https://twitter.com/allthebestword1"
                      >
                        Visit Project Vobot
                      </a>
                      <a
                        class="gitLink"
                        href="https://github.com/VTMFOA"
                      >
                        Git Repo
                      </a>
                    </div>
                  </li>
                  <li>
                    <div
                      class="projectItem"
                    >
                      <h2
                        class="name"
                      >
                        What's Cook'n
                      </h2>
                      <img
                        alt="What's Cook'n"
                        class="projectPicInView"
                        src="/assets/whatscookn.png"
                      />
                      <p
                        class="description"
                      >
                        A meal planner complete with recipe finder.
                      </p>
                      <a
                        class="siteLink"
                        href="https://whats-cookn.netlify.app/"
                      >
                        Visit What's Cook'n
                      </a>
                      <a
                        class="gitLink"
                        href="https://github.com/Aluminum-Chefs"
                      >
                        Git Repo
                      </a>
                    </div>
                  </li>
                </ul>
                <div
                  class="arrowBox"
                >
                  <div
                    class="arrowBox"
                  >
                    <div
                      class="arrow"
                    >
                      ï¸¾
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="listBox2"
          >
            <div
              class="projectBox"
              style="transition: ease-in; visibility: visible;"
            >
              <div
                class="listBox"
              >
                <h1
                  class="headline"
                >
                  Mobile Projects
                </h1>
                <ul
                  class="projectList"
                >
                  <li>
                    <div
                      class="projectItem"
                    >
                      <h2
                        class="name"
                      >
                        nerdMeme
                      </h2>
                      <img
                        alt="nerdMeme"
                        class="mobileProjectPic"
                        src="/assets/nerdMemePic.png"
                      />
                      <p
                        class="description"
                      >
                        A random meme generator.
                      </p>
                      <a
                        class="downloadLink"
                        download=""
                        href="/assets/nerdmeme-1-2.apk"
                      >
                        Download nerdMeme
          for Android
                      </a>
                      <a
                        class="gitLink"
                        href="https://github.com/Meme-Nerds"
                      >
                        Git Repo
                      </a>
                    </div>
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            class="arrowBox"
          >
            <div
              class="arrowBox"
            >
              <div
                class="arrow"
              >
                ï¸¾
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sectionTwo"
      >
        <div
          class="workPageBox"
          id="freelance"
        >
          <div
            class="bigBox"
          >
            <h1
              class="proHeadline"
            >
              Freelance Work
            </h1>
            <div
              class="services"
            >
              <h3
                class="invisible"
              >
                if you want to . . .
              </h3>
              <h4
                class="invisible"
              >
                expand your digital brand,
              </h4>
              <h4
                class="invisible"
              >
                manage your data,
              </h4>
              <h4
                class="invisible"
              >
                or build a cool app,
              </h4>
              <h3
                class="invisible"
              >
                . . . I can help.
              </h3>
            </div>
            <div
              class="workBox"
            >
              <h1
                class="subHeadline"
              >
                Amphead/Lanham
              </h1>
              <div
                class="picBox"
              >
                <img
                  class="workPic"
                  src="/assets/ampheadScreenShot.png"
                />
                <p
                  class="workDescription"
                >
                  A full-stack web app for a multi-faceted Portland business. Amphead services musical equipment and Lanham builds custom Guitar Amplifiers. One website that acts as two. A NodeJS/Express backend API with PostgreSQL database, serves the data. A React/Redux front end brings it all to the consumer. Protected routes with full authorization allow the site owner to manage content through an administrator UI.
                </p>
                <a
                  href="https://loving-gates-9e8b10.netlify.app/"
                >
                  visit the site
                </a>
                <a
                  href="https://github.com/Amphead-Lanham"
                >
                  git repo
                </a>
              </div>
            </div>
            <div
              class="workBox2"
            >
              <div
                class="picBox"
              >
                <img
                  class="workPic"
                  src="/assets/MAAScreenShot.png"
                />
                <p
                  class="workDescription"
                >
                  An artist's gallery web app with a React/TypeScript/Redux front-end and a NodeJS back-end that utilizes the Cloudinary API. The site owner can upload their own high quality images and manage all content with an admin UI through a protected route. With a mobile-first design, this app looks great on all devices.
                </p>
                <a
                  href="https://www.michaelalbano.art/"
                >
                  visit the site
                </a>
                <a
                  href="https://github.com/Michael-Albano-Artist"
                >
                  git repo
                </a>
              </div>
              <h1
                class="subHeadline2"
              >
                michaelalbano.art
              </h1>
            </div>
            <div
              class="arrowBox"
            >
              <div
                class="arrowBox"
              >
                <div
                  class="arrow"
                >
                  ï¸¾
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sectionThree"
      >
        <div
          class="techPage"
          id="tech"
        >
          <div
            class="techPageInner"
          >
            <h1
              class="techHeadline"
            >
              Tech-Stack
            </h1>
            <div
              class="midDisplay"
            >
              <div>
                <div
                  class="oneTech"
                >
                  <div
                    class="componentBox"
                  >
                    <ul
                      class="animatedTechTypeyes"
                    >
                      <li>
                        <h1
                          class="techCategory"
                        >
                          languages
                        </h1>
                      </li>
                      <li
                        class="tech0"
                      >
                        <h2>
                          JavaScript
                        </h2>
                      </li>
                      <li
                        class="tech1"
                      >
                        <h2>
                          TypeScript
                        </h2>
                      </li>
                      <li
                        class="tech2"
                      >
                        <h2>
                          CSS
                        </h2>
                      </li>
                      <li
                        class="tech3"
                      >
                        <h2>
                          HTML
                        </h2>
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="tOverview"
            >
              <div
                class="overview"
              >
                <p>
                  ã
                </p>
                <div>
                  Front-End,
                </div>
                <div>
                  Back-End,
                </div>
                <div>
                  Responsive-Design
                </div>
                <p>
                  ã
                </p>
              </div>
            </div>
            <div
              class="arrowBox"
            >
              <div
                class="arrowBox"
              >
                <div
                  class="arrow"
                >
                  ï¸¾
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sectionFour"
      >
        <div
          class="bio"
          id="about"
        >
          <h1>
            A Bit More About Me
          </h1>
          <div>
            <p>
              ï½
            </p>
            <h3
              class="bioText"
            >
              I am a Full-Stack software developer living in Portland, OR.  I love creating things with code from end to end.  My starter career was bartending and from there I moved into small business ownership. I have always had a focus on customer service and delivering for the the client.  I'm an artist and a musician and really enjoy bringing that type of creative energy to tech.  I love to spend my spare time with my family, whether that be at home or out in the beauty of Oregon.
            </h3>
            <p>
              ï½
            </p>
          </div>
        </div>
      </div>
      <div
        class="sectionFive"
      >
        <div
          class="bioPage"
        >
          <section>
            <div
              class="contact"
            >
              <div
                class="contactBox"
              >
                <div
                  class="elementBox"
                >
                  <div
                    class="popUpInner"
                  >
                    <div>
                      <div
                        class="popUp"
                      >
                        <label
                          class="label"
                          for="form-checkboxbio"
                        >
                          <img
                            alt="email"
                            class="pic"
                            src="/path/image"
                          />
                        </label>
                        <input
                          class="toggle"
                          id="form-checkboxbio"
                          type="checkbox"
                        />
                        <div
                          class="contain"
                        >
                          <div />
                        </div>
                        <div
                          class="form"
                        >
                          <div>
                            <div>
                              <div
                                class="contact"
                              >
                                <label
                                  class="Xbox"
                                  for="form-checkboxbio"
                                >
                                  X
                                </label>
                                <input
                                  class="checkbox"
                                  id="form-checkboxbio"
                                  type="checkbox"
                                />
                                <form
                                  id="contact-formbio"
                                  method="POST"
                                >
                                  <input
                                    id="namebio"
                                    name="name"
                                    placeholder="name required"
                                    type="text"
                                    value=""
                                  />
                                  <input
                                    id="emailbio"
                                    name="email"
                                    placeholder="email required"
                                    type="text"
                                    value=""
                                  />
                                  <textarea
                                    id="messagebio"
                                    name="message"
                                    placeholder="message required"
                                  />
                                  <button>
                                    submit
                                  </button>
                                </form>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="elementBox"
                >
                  <a
                    href="/assets/jerudresume.pdf"
                  >
                    <img
                      alt="PDF"
                      class="mediaPic"
                      src="/path/image"
                    />
                  </a>
                </div>
                <div
                  class="elementBox"
                >
                  <a
                    href="https://twitter.com/Jerud7"
                  >
                    <img
                      alt="Twitter"
                      class="mediaPic"
                      src="/path/image"
                    />
                  </a>
                </div>
                <div
                  class="elementBox"
                >
                  <a
                    href="https://www.linkedin.com/in/jerud-moyer/"
                  >
                    <img
                      alt="LinkedIn"
                      class="mediaPic"
                      src="/path/image"
                    />
                  </a>
                </div>
                <div
                  class="elementBox"
                >
                  <a
                    href="https://github.com/Jerud-Moyer"
                  >
                    <img
                      alt="GitHub"
                      class="mediaPic"
                      src="/path/image"
                    />
                  </a>
                </div>
              </div>
            </div>
            <div>
              <div
                class="picFlip"
              >
                <div
                  class="frame"
                >
                  <div>
                    <img
                      alt="Autumn in Portland"
                      class="picActive"
                      src="https://i.imgur.com/kyoJOLj.jpg"
                    />
                    <h4
                      class="caption"
                    >
                      Autumn in Portland
                    </h4>
                  </div>
                  <div />
                  <div />
                  <div />
                  <div />
                  <div />
                  <div />
                  <div />
                  <div />
                  <div />
                  <div />
                  <div />
                  <div />
                  <div />
                  <div />
                  <div />
                  <div />
                </div>
                <section>
                  <div>
                    ã
                  </div>
                  <div
                    class="indicatorActive"
                  />
                  <div
                    class="indicator"
                  />
                  <div
                    class="indicator"
                  />
                  <div
                    class="indicator"
                  />
                  <div
                    class="indicator"
                  />
                  <div
                    class="indicator"
                  />
                  <div
                    class="indicator"
                  />
                  <div
                    class="indicator"
                  />
                  <div
                    class="indicator"
                  />
                  <div
                    class="indicator"
                  />
                  <div
                    class="indicator"
                  />
                  <div
                    class="indicator"
                  />
                  <div
                    class="indicator"
                  />
                  <div
                    class="indicator"
                  />
                  <div
                    class="indicator"
                  />
                  <div
                    class="indicator"
                  />
                  <div
                    class="indicator"
                  />
                  <div>
                    ã
                  </div>
                </section>
              </div>
            </div>
            <aside>
              <nav
                class="nav"
              >
                <h3>
                  music
                </h3>
                <a
                  href="https://dinnerforwolves.bandcamp.com/"
                >
                  Dinner for Wolves
                </a>
                <a
                  href="https://dirtylowdowns.bandcamp.com/"
                >
                  Dirty Lowdowns
                </a>
              </nav>
              <nav>
                <a
                  href="/list"
                >
                  random facts about me
                </a>
                <div>
                  back to top
                </div>
              </nav>
            </aside>
          </section>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
